src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -89,44 +111,53 @@ public static DeclarationModifiers GetSymbolModifiers(this ISymbol symbol)
         }
 
-        private static bool IsBrowsingProhibited(
+        private static (bool isProhibited, bool isEditorBrowsableStateAdvanced) IsBrowsingProhibited(
             ISymbol symbol,
             bool hideAdvancedMembers,
             Compilation compilation,
-            IMethodSymbol? editorBrowsableAttributeConstructor,
-            List<IMethodSymbol>? typeLibTypeAttributeConstructors,
-            List<IMethodSymbol>? typeLibFuncAttributeConstructors,
-            List<IMethodSymbol>? typeLibVarAttributeConstructors,
-            INamedTypeSymbol? hideModuleNameAttribute)
+            Optional<IMethodSymbol?> editorBrowsableAttributeConstructor,
+            ImmutableArray<IMethodSymbol> typeLibTypeAttributeConstructors,
+            ImmutableArray<IMethodSymbol> typeLibFuncAttributeConstructors,
+            ImmutableArray<IMethodSymbol> typeLibVarAttributeConstructors,
+            Optional<INamedTypeSymbol?> hideModuleNameAttribute)
         {
             var attributes = symbol.GetAttributes();
             if (attributes.Length == 0)
             {
-                return false;
+                return (isProhibited: false, isEditorBrowsableStateAdvanced: false);
             }
 
-            return IsBrowsingProhibitedByEditorBrowsableAttribute(attributes, hideAdvancedMembers, compilation, editorBrowsableAttributeConstructor)
+            var (isProhibited, isEditorBrowsableStateAdvanced) = IsBrowsingProhibitedByEditorBrowsableAttribute(attributes, hideAdvancedMembers, compilation, editorBrowsableAttributeConstructor);
+
+            return ((isProhibited
                 || IsBrowsingProhibitedByTypeLibTypeAttribute(attributes, compilation, typeLibTypeAttributeConstructors)
                 || IsBrowsingProhibitedByTypeLibFuncAttribute(attributes, compilation, typeLibFuncAttributeConstructors)
                 || IsBrowsingProhibitedByTypeLibVarAttribute(attributes, compilation, typeLibVarAttributeConstructors)
-                || IsBrowsingProhibitedByHideModuleNameAttribute(symbol, compilation, hideModuleNameAttribute, attributes);
+                || IsBrowsingProhibitedByHideModuleNameAttribute(symbol, compilation, hideModuleNameAttribute, attributes)), isEditorBrowsableStateAdvanced);
         }
 
         private static bool IsBrowsingProhibitedByHideModuleNameAttribute(
-            ISymbol symbol, Compilation compilation, INamedTypeSymbol? hideModuleNameAttribute, ImmutableArray<AttributeData> attributes = default)
+            ISymbol symbol, Compilation compilation, Optional<INamedTypeSymbol?> hideModuleNameAttribute, ImmutableArray<AttributeData> attributes = default)
         {
             if (!symbol.IsModuleType())
             {
                 return false;
             }
 
             attributes = attributes.IsDefault ? symbol.GetAttributes() : attributes;
-            hideModuleNameAttribute ??= compilation.HideModuleNameAttribute();
+
+            if (!hideModuleNameAttribute.HasValue)
```

Comment:
afaiut, this will never be false.  so we'll never go into this branch.

Reply:
`IsEditorBrowsable` can be called with default parameter value, in which case `HasValue` will be false.

Reply:
basically, my preference woudl be that instead of taking N extra parameters, you take an optional EBInfo struct (for caching those values).  If that value is not provided, you instantiate a new one in this method.

Reply:
would it be hard to take an optional EBInfo?

Reply:
I felt it might be unnecessary given it's only called from two locations (that provides value for those optional parameters). But sure.